check_limits.py

```diff
@@ -1,18 +1,43 @@
 
-def battery_is_ok(temperature, soc, charge_rate):
+def battery_is_ok(temperature, soc, charge_rate):     
+  status_list = [temperature_in_range(temperature), soc_in_range(soc), charge_rate_in_range(charge_rate)]
+  for statuses in status_list:
+    if statuses == False:
+      return False
+  return True
+
+
+def temperature_in_range(temperature):
   if temperature < 0 or temperature > 45:
```

Comment:
similar code is present in SOC check too. There is scope to reduce code duplication.